core/backend/src/ECSqlStatement.ts

```diff
@@ -320,36 +322,45 @@ export class ECSqlStatement implements IterableIterator<any>, IDisposable {
    * - [ECSQL row format]($docs/learning/ECSQLRowFormat) for details about the format of the returned row.
    * - [Code Samples]($docs/learning/backend/ECSQLCodeExamples#working-with-the-query-result)
    */
-  public getRow(): any {
-    const colCount: number = this.getColumnCount();
-    const row: object = {};
-    const duplicatePropNames = new Map<string, number>();
-    for (let i = 0; i < colCount; i++) {
-      const ecsqlValue = this.getValue(i);
-      if (!ecsqlValue.isNull) {
-        const propName: string = ECSqlStatement.determineResultRowPropertyName(duplicatePropNames, ecsqlValue);
-        const val: any = ecsqlValue.value;
-        Object.defineProperty(row, propName, { enumerable: true, configurable: true, writable: true, value: val });
-      }
+  public getRow(args?: IModelJsNative.ECSqlRowArg): any {
+    if (!args) {
+      args = { abbreviateBlobs: false, convertClassIdsToClassNames: true, rowFormat: QueryRowFormat.UseJsPropertyNames, includeMetaData: this._props.length === 0 };
     }
-    return row;
+    const resp = this._stmt!.toRow(args); // eslint-disable-line @typescript-eslint/no-non-null-assertion
```

Comment:
Previous `getRow` implementation used to assert stmt is not undefined and didn't call the native API, as an error would be thrown. This now seems to call the new getRow function always regardless

Reply:
Earlier `getRow` function was used to call the `getColumnCount()` function, which ultimately called the native API similar to our current approach.
In `core\backend\src\ECSqlStatement.ts`, there are a few places where we've disabled the `@typescript-eslint/no-non-null-assertion` rule.